src/index.ts

```diff
@@ -45,29 +45,27 @@ export default (apiUrl, httpClient = fetchUtils.fetchJson): DataProvider => ({
         };
         const url = `${apiUrl}/records/${resource}?${stringify(query)}`;
 
-        return httpClient(url).then(({ json }) => ({
+        const { json } = await httpClient(url);
+        return ({
             data: json.records,
-            total: json.results,
-        }));
+            total: json.results ?? 0,
+        });
     },
 
-    getOne: (resource, params) => {
-        return httpClient(`${apiUrl}/records/${resource}/${params.id}`).then(({ json }) => ({
+    getOne: async (resource, params) => {
+        const { json } = await httpClient(`${apiUrl}/records/${resource}/${params.id}`);
+        return ({
             data: json,
-        }));
+        });
     },
 
-    getMany: (resource, params) => {
+    getMany: async (resource, params) => {
         const url = `${apiUrl}/records/${resource}/${params.ids.join(',')}`;
-        if (params.ids.length === 1) {
-            return httpClient(url).then(({ json }) => ({ data: [json] })); // needs to be an array even if one id is passed
-        } else {
-            return httpClient(url).then(({ json }) => ({ data: json }));
-        }
+        return httpClient(url).then(({ json }) => ({ data: Array.isArray(json) ? json : [json] }));
```

Comment:
just noticed that the auto fixes missed some parts and can also be made simple... I'll push another revision soon...